pennylane_lightning/_serialize.py

```diff
@@ -58,162 +40,210 @@
 }
 
 
-def _serialize_named_obs(ob, wires_map: dict, use_csingle: bool):
-    """Serializes a Named observable"""
-    named_obs = NamedObsC64 if use_csingle else NamedObsC128
-    wires = [wires_map[w] for w in ob.wires]
-    if ob.name == "Identity":
-        wires = wires[:1]
-    return named_obs(ob.name, wires)
-
-
-def _serialize_hermitian_ob(o, wires_map: dict, use_csingle: bool):
-    """Serializes a Hermitian observable"""
-    assert not isinstance(o, Tensor)
-
-    if use_csingle:
-        ctype = np.complex64
-        hermitian_obs = HermitianObsC64
-    else:
-        ctype = np.complex128
-        hermitian_obs = HermitianObsC128
-
-    wires = [wires_map[w] for w in o.wires]
-    return hermitian_obs(matrix(o).ravel().astype(ctype), wires)
-
+class _Serialize:
+    def __init__(self, device_name):
+        if device_name == "lightning.qubit":
+            try:
+                from .lightning_qubit_ops import (
+                    StateVectorC128,
+                )
+                from .lightning_qubit_ops.observables import (
+                    NamedObsC64,
+                    NamedObsC128,
+                    HermitianObsC64,
+                    HermitianObsC128,
+                    TensorProdObsC64,
+                    TensorProdObsC128,
+                    HamiltonianC64,
+                    HamiltonianC128,
+                )
+            except ImportError:
+                raise ImportError(
+                    "Pre-compiled binaries for "
+                    + device_name
+                    + " serialize functionality are not available."
+                )
+        elif device_name == "lightning.kokkos":
+            try:
+                from .lightning_kokkos_ops import (
+                    StateVectorC128,
```

Comment:
Is this missing?
```suggestion
                    StateVectorC64,
                    StateVectorC128,
```

Reply:
Yes, but I think it would be cleaner to have a `state_vector` property returning the correct binding depending on `use_csingle` at [line 237](https://github.com/PennyLaneAI/Lightning-Unification/pull/21/files/2f95160524d62d1ed9681c44e1784af8721553d6#diff-e157fb0539b95daab56a38c551225146c37e1ae1580811fa3df65f73e50dc905R237); and then you would need both.

Reply:
I don't think this is necessary, not with the use we have in place for `StateVectorC128`.